cxr-foundation/cxr_foundation/embeddings_data.py

```diff
@@ -0,0 +1,167 @@
+        value: "some/long/imageid"
+      }
+    }
+  }
+}
+"""
+from typing import Iterable
+
+import numpy as np
+import tensorflow as tf
+
+from cxr_foundation import constants
+
+
+def read_record_example(filename: str) -> tf.train.Example:
+  """
+  Read the tf.Example data contained in a single TFRecord embedding file.
+
+  Args:
+    filename: The path of the .tfrecord file to read
+
+  Returns:
+    The `tf.Example` data contained in the TFRecord.
+
+  Note: This is a convenience function for exploring/exporting. Do not use this in TF pipelines.
+
+  """
+  raw_dataset = tf.data.TFRecordDataset(filename)
+  # Expect only one element in the TFRecord.
+  for raw_record in raw_dataset.take(1):
+    example = tf.train.Example()
+    example.ParseFromString(raw_record.numpy())
+
+  return example
+
+def extract_embedding_values(example_data: tf.train.Example) -> np.ndarray:
+  """
+  Extract the embeddings values contained in an Example object, extracted from a file
+  generated by the CXR foundation service.
+
+  Args:
+    example_data: The Example object to extract the values from
+
+  Returns:
+    The 1D float array of the embeddings values
+
+  Note: This is a convenience function for exploring/exporting. Do not use this in TF pipelines.
+
+  """
+  # Unpack nested proto and create np array from google.protobuf.pyext._message.RepeatedScalarContainer
+  try:
+    values = np.array(example_data.features.feature[constants.EMBEDDING_KEY].float_list.value, dtype='int32')
+    return values
+  except ValueError:
+    print(f"Input Example does not contain expected CXR Foundation embedding structure.")
+    raise
+
+def read_record_embeddings(filename: str) -> np.ndarray:
+  """
+  Read the embeddings values contained in a TFRecord embedding file, generated by the CXR
+  foundation service.
+
+  Args:
+    filename: The path of the .tfrecord file to read
+
+  Returns:
+    The 1D float array of the embeddings values
+
+  Note: This is a convenience function for exploring/exporting. Do not use this in TF pipelines.
+
+  """
+  return extract_embedding_values(read_record_example(filename))
+
+
+def parse_embedding_from_example(
+    serialized_example: bytes,
+) -> tf.Tensor:
+  """
+  Parses and extracts the embeddings values from a serialized tf.Example generated by the CXR foundation service.
+
+  Args:
+    serialized_example: The bytes of the tf.Example to parse.
+
+  Returns:
+    The 1D Tensor of float embeddings
+
+  """
+  features = {
+      constants.EMBEDDING_KEY:
+          tf.io.FixedLenFeature([constants.DEFAULT_EMBEDDINGS_SIZE],
+                                tf.float32,
+                                default_value=tf.constant(
+                                    0.0, shape=[constants.DEFAULT_EMBEDDINGS_SIZE]))
+  }
+  parsed_tensors = tf.io.parse_example(serialized_example, features=features)
+  return parsed_tensors[constants.EMBEDDING_KEY]
+
+
+def get_dataset(
+    filenames: Iterable[str],
+    labels: Iterable[int]
```

Comment:
Not sure how this does performance wise, relative to the previous creation of the static hash map from the input labels dict.